pkg/server/providers.go

```diff
@@ -88,6 +104,26 @@ func (s *Server) registerProviders() error {
 	return nil
 }
 
+func (s *Server) lockInitialProviderSetups() error {
```

Comment:
Why is this required? Why don't we simply write the lock file after a provider was downloaded in `downloadDefaultProviders()`?

Reply:
> Okay, lets at least go with this (https://github.com/daytonaio/daytona/pull/1288#discussion_r1821562325) then.

If we were to write the lock file after downloading, then the `RegisterProviders` function would find it and not run `setPresetTargets`

Reply:
Can we write it in `registerProviders` then?

Reply:
Okay yes we can, it's done in `RegisterProvider` now